crates/ruff/src/rules/pylint/rules/yield_in_init.rs

```diff
@@ -14,13 +14,16 @@ use crate::{
 define_violation!(
     /// ### What it does
     /// Checks for `__init__` methods that are turned into generators by the
-    /// inclusion of `yield` or `yield from` statements.
+    /// inclusion of `yield` or `yield from` expressions.
     ///
     /// ### Why is this bad?
-    /// The `__init__` method of a class is used to initialize new objects, not
-    /// create them. As such, it should not return any value. By including a
-    /// yield expression in the method turns it into a generator method. On
-    /// calling, it will return a generator resulting in a runtime error.
+    /// The `__init__` method is the constructor for a given Python class,
+    /// responsible for initializing, rather than creating, new objects.
+    ///
+    /// The `__init__` method is not allowed to return a value. By including a
```

Comment:
I think the first sentence should be changed to:

> The `__init__` method has to return `None`.

Since I'd argue that `None` is technically also a value.

Reply:
Yeah I had that initially but it might imply to some that you have to _explicitly_ `return None`. Do you think it's clearer?

Reply:
Yeah I think it's clearer, quoting the [control flow page of the Python tutorial](https://docs.python.org/3/tutorial/controlflow.html):

> In fact, even functions without a [return](https://docs.python.org/3/reference/simple_stmts.html#return) statement do return a value, albeit a rather boring one. This value is called None (it’s a built-in name).

Yes this wording might be confusing for users not aware of this implicit `None` return but I think it's better to have accurate wording.